ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/assertions/CommandDeduplicationAssertions.scala

```diff
@@ -51,37 +55,93 @@ object CommandDeduplicationAssertions {
-          WithTimeout(5.seconds)(
-            ledger
-              .findCompletion(completionStreamRequest)(c =>
-                c.commandId == c.commandId && c.getStatus.code == Code.OK.value
-              )
-          ).map { optAcceptedCompletionResponse =>
-            val acceptedCompletionResponse = assertDefined(
-              optAcceptedCompletionResponse,
-              s"No accepted completion with the command ID '${completionResponse.completion.commandId}' has been found",
-            )
-            assert(
-              acceptedCompletionResponse.offset.getAbsolute >= reportedOffset,
-              s"No accepted completion with the command ID '${completionResponse.completion.commandId}' after the reported offset $reportedOffset has been found",
-            )
-            assert(
-              acceptedCompletionResponse.offset.getAbsolute < completionResponse.offset.getAbsolute,
-              s"No accepted completion with the command ID '${completionResponse.completion.commandId}' before the completion's offset ${completionResponse.offset} has been found",
-            )
-          }
+          assertReportedOffsetForDuplicateSubmission(
+            reportedHexOffset,
+            completionResponse,
+            ledger,
+            reportedOffsetCompletionStreamRequest,
+          )
         } else {
-          fail("This case is not yet supported by this assertion")
+          assertReportedOffsetForAcceptedSubmission(
+            reportedHexOffset,
+            requestedDuration,
+            completionResponse,
+            ledger,
+            reportedOffsetCompletionStreamRequest,
+          )
         }
       case DurationSupport.Unrecognized(_) =>
         fail("Unrecognized deduplication duration support")
     }
   }
 
+  private def assertReportedOffsetForDuplicateSubmission(
+      reportedOffset: Ref.HexString,
+      completionResponse: CompletionResponse,
+      ledger: ParticipantTestContext,
+      reportedOffsetCompletionStreamRequest: CompletionStreamRequest,
+  )(implicit executionContext: ExecutionContext) = {
+    WithTimeout(5.seconds)(
+      ledger
+        .findCompletion(reportedOffsetCompletionStreamRequest)(c =>
+          c.commandId == completionResponse.completion.commandId && c.getStatus.code == Code.OK.value
+        )
+    ).map { optAcceptedCompletionResponse =>
+      val acceptedCompletionResponse = assertDefined(
+        optAcceptedCompletionResponse,
+        s"No accepted completion with the command ID '${completionResponse.completion.commandId}' since the reported offset $reportedOffset has been found",
+      )
+      assert(
+        acceptedCompletionResponse.offset.getAbsolute < completionResponse.offset.getAbsolute,
+        s"No accepted completion with the command ID '${completionResponse.completion.commandId}' before the completion's offset ${completionResponse.offset} has been found",
```

Comment:
I'm not sure we can, @nicu-da, @fabiotudone-da. This would effectively mean that the previous/accepted completion is the same as the duplicate one that we assert. cc @andreaslochbihler-da

Reply:
I think @nicu-da just wants to have the accepted offset be part of the error message.

Reply:
Ok, the inclusivity/exclusivity hit me too hard, apparently.

Reply:
Done